app/src/main/java/ca/marshallasch/veil/database/Migrations.java

```diff
@@ -209,6 +209,75 @@ static void upgradeV9(SQLiteDatabase db) {
      * @param db the database object
      */
     static void upgradeV10(SQLiteDatabase db){
+
+        class KnownPostObject{
+            String postHash;
+            String commentHash;
+            int readStatus;
+            Long insertTime;
+
+            KnownPostObject(String _postHash, String _commentHash, int _readStatus, Long _insertTime){
+                postHash = _postHash;
+                commentHash = _commentHash;
+                readStatus = _readStatus;
+                insertTime = _insertTime;
+            }
+        }
+
+        List<KnownPostObject> knownPosts = new ArrayList<>();
+
+        String[] projection = {
+                KnownPostsContract.KnownPostsEntry.COLUMN_POST_HASH,
+                KnownPostsContract.KnownPostsEntry.COLUMN_COMMENT_HASH,
+                KnownPostsContract.KnownPostsEntry.COLUMN_READ,
+                KnownPostsContract.KnownPostsEntry.COLUMN_TIME_INSERTED
+        };
+
+
+        Cursor cursor = db.query(
+                "KnownPostsContract",   // The table to query
```

Comment:
The table name should be `KnownPostsContract.KnownPostsEntry.TABLE_NAME` This is what is causing the migration crashes.

Reply:
resolved